File: 1erdepartamental/c++/9.promedio_arreglo/main.cpp
#include <iostream>
#include "core/EstadisticasArreglo.h"

int main() {
    // Crea estadisticas con inicializer list
    EstadisticasArreglo e({10, 25, 5, 30, 15});
    
    // Muestra estadisticas iniciales
    std::cout << "Datos iniciales: " << e.getDatos().size() << " elementos" << std::endl;
    std::cout << "Suma: " << e.suma() << std::endl;
    std::cout << "Promedio: " << e.promedio() << std::endl;
    std::cout << "Máximo: " << e.maximo() << std::endl;
    std::cout << "Mínimo: " << e.minimo() << std::endl << std::endl;
    
    // Agrega un valor y recalcula
    e.agregar(50);
    std::cout << "Después de agregar 50:" << std::endl;
    std::cout << "Promedio: " << e.promedio() << std::endl;
    
    return 0;
}

File: 1erdepartamental/python/9.promedio_arreglo/main.py
"""
Programa: Promedio con Arreglo
Demuestra estadísticas con arreglo dinámico
"""
#Importa la clase 
from core.estadisticas import EstadisticasArreglo
#Punto de entrada del programa
if __name__ == "__main__":
    e = EstadisticasArreglo([10, 25, 5, 30, 15])

    print(f"Suma: {e.suma()}")
    print(f"Promedio: {e.promedio()}")
    print(f"Máximo: {e.maximo()}")
    print(f"Mínimo: {e.minimo()}\n")

    e.agregar(50)
    print(f"Después de agregar 50:")
    print(f"Promedio: {e.promedio()}")
